jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/ContextHandlerManagement.java

```diff
@@ -0,0 +1,32 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.deploy;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.ContextHandler;
+
+/**
+ * Deployment Manager specific methods that manage ContextHandler.
+ */
+public interface ContextHandlerManagement
+{
+    Server getServer();
+
+    // TODO: document methods
+    void addContextHandler(ContextHandler contextHandler, String goalName);
```

Comment:
Maybe the method signature could be `String addContextHandler(ContextHandler)` where a deployment id is returned that can be used to uniquely reference that `ContextHandler` in `DeploymentManager` methods, eg in a `removeContextHandler(String id)` method.

Reply:
The ID is controlled by the bean calling this method, not DeploymentManager.

Reply:
The ID is only relevant to fact that the `ContextHandler` is being deployed, so not something that the "providers" need to worry about. They just need to retain it so they can later on refer to it when interacting with the `DeploymentManager`.

Reply:
The older "providers" (should really be called "ContextHandler Source") is the one that assigns the ID, and can use the ID to manipulate the ContextHandler through the ContextHandlerLifeCycle in the DeploymentManager.

Other ContextHandler Source's don't know about, or care about, ContextHandlers that they don't manage.

Tooling that interacts directly with DeploymentManager (eg: JMX) is where the ID shines.

Reply:
> The ID is controlled by the bean calling this method, not DeploymentManager.

How do we avoid collisions in this case?    What if two providers (say for different virtual hosts) both have a "/" context they want to call "root"?      The DM will think these are the same CH.

I think if we want a unique ID for a CH, then either it needs to be a URI to something OR the DM needs to manage it.